composer/loggers/slack_logger.py

```diff
@@ -0,0 +1,143 @@
+        max_logs_per_message (int)(default:50): Maximum number of logs to send in a single message. Note that no more than 50 items are allowed to send in a single message.
+        If more than 50 items are stored in buffer, the message flushed without waiting the full time interval.
+    """
+
+    def __init__(
+        self,
+        formatter_func: Optional[Callable[..., List[Dict[str, Any]]]] = None,
+        include_keys: Optional[Set[str]] = None,
+        interval_in_seconds: Optional[int] = None,
+        max_logs_per_message: int = 50,
+    ) -> None:
+        try:
+            import slack_sdk
+            self.client = slack_sdk.WebClient()
+            del slack_sdk
+        except ImportError as e:
+            raise MissingConditionalImportError('slack_logger', 'slack_sdk', None) from e
+
+        self.slack_logging_api_key = os.environ.get('SLACK_LOGGING_API_KEY', None)
+        self.channel_id = os.environ.get('SLACK_LOGGING_CHANNEL_ID', None)
+
+        if self.slack_logging_api_key is None:
+            raise RuntimeError('SLACK_LOGGING_API_KEY must be set as environment variable')
+        if self.channel_id is None:
+            raise RuntimeError('SLACK_LOGGING_CHANNEL_ID must be set as environment variable')
+
+        self.formatter_func = formatter_func
+        self.include_keys = include_keys
+        self.interval_in_seconds = interval_in_seconds
+
+        self.logs, self.last_log_time = [], time.time()
+        self.max_logs_per_message = min(max_logs_per_message, 50)
+
+    # Rich message layouts can be created using Slack Blocks Kit.
+    # See documentation here: https://api.slack.com/messaging/composing/layouts
+    def _log_to_slack(
+            self,
+            data: Dict[str, Any],
+            **kwargs,  # can be used to pass additional arguments to the formatter function (eg for headers)
+    ):
+        filtered_data = {k: v for k, v in data.items() if k in self.include_keys
+                        } if self.include_keys is not None else data
+        blocks = self.formatter_func(
+            filtered_data, **
+            kwargs) if self.formatter_func is not None else self._default_log_bold_key_normal_value_pair_with_header(
+                filtered_data, **kwargs)
+
+        self.logs += blocks
+        now = time.time()
+
+        if len(self.logs) > 0 and self.channel_id is not None and (
+                len(self.logs) >= self.max_logs_per_message or self.interval_in_seconds is None or
+                now - self.last_log_time >= self.interval_in_seconds):
+            self.client.chat_postMessage(token=f'{self.slack_logging_api_key}',
+                                         channel=self.channel_id,
+                                         blocks=self.logs[:self.max_logs_per_message],
+                                         text=str(self.logs[:self.max_logs_per_message]))
```

Comment:
the `text` variable is supposed to be the formatted text in either plaintext or markdown right? it's not supposed be just a stringification of the block right? https://api.slack.com/methods/chat.postMessage#blocks_and_attachments

Reply:
`When you're using blocks in your message payload, the top-level text field becomes a fallback message displayed in notifications. Blocks should define everything else about the desired visible message.` 

text is there for notifications actually. I didn't test this behavior manually, but I definitely should, and probably just updated to a shorter string like f"Value for {len(self.logs)} metrics updated"